src/model/chord.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{Pitch, Quality};
+use combine::{Parser, Stream};
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Chord {
+    pub pitch: Pitch,
+    pub quality: Quality,
+}
+
+impl Chord {
+    pub fn new(pitch: Pitch, quality: Quality) -> Self {
+        Self { pitch, quality }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        (Pitch::parser(), Quality::parser()).map(|(pitch, quality)| Chord::new(pitch, quality))
+    }
+}
+
+impl fmt::Display for Chord {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&format!("{}{}", self.pitch, self.quality))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn parser() {
+        let chords: Vec<_> = Pitch::all()
+            .into_iter()
+            .flat_map(|p| Quality::iter().map(move |q| Chord::new(p, q)))
```

Comment:
I'm trying to decide if I mind your p's and q's (as far as variable naming - I think I'm not a fan of one-letter variable names even for closure arguments like this, I'd prefer `pitch` + `quality`)

Reply:
Updating with real names.

> I'm trying to decide if I mind your p's and q's

I bet you are pretty pleased with yourself!